packages/core/src/indices/vectorStore/VectorStoreIndex.ts

```diff
@@ -276,9 +275,11 @@ export class VectorStoreIndex extends BaseIndex<IndexDict> {
       if (
         !vectorStore.storesText ||
         type === ObjectType.INDEX ||
-        type === ObjectType.IMAGE
+        type === ObjectType.IMAGE ||
+        type === ObjectType.IMAGE_DOCUMENT
```

Comment:
Just curious, what's the difference between IMAGE and IMAGE_DOCUMENT ? Is IMAGE a node?

Reply:
Yes exactly and `IMAGE_DOCUMENT` is a document containing an image. The image loader generates type `IMAGE_DOCUMENT`, so I had to make this change here